Nodejs/Product/Analysis/JavaScript/declaration.cs

```diff
@@ -56,7 +56,8 @@ public override IEnumerable<Node> Children
 
             set {
                 for (int i = 0; i < value.Length; i++) {
-                    value[i].Parent = this;
+                    if (value[i] != null)
```

Comment:
Also, in general, please include braces with if statements (even if they are single-statement) because it keeps the code more readable and maintainable.